ops/terraform/modules/mgmt_stateful/main.tf

```diff
@@ -121,33 +111,54 @@ resource "aws_iam_policy" "ansible_vault_pw_ro_s3" {
 EOF
 }
 
-# S3 bucket for Build Artifacts
+# S3 Admin bucket for adminstrative stuff
 #
-module "artifacts" {
+
+module "admin" { 
   source              = "../resources/s3"
-  role                = "artifacts"
+  role                = "admin"
   env_config          = local.env_config
   kms_key_id          = data.aws_kms_key.master_key.arn
-  log_bucket          = module.admin.id
+  log_bucket          = module.logs.id
+}
+
+# S3 bucket for logs 
+#
+
+module "logs" { 
+  source              = "../resources/s3"
+  role                = "logs"
+  env_config          = local.env_config
+  acl                 = "log-delivery-write"  # For AWS bucket logs
+  kms_key_id          = null                  # Use AWS encryption to support AWS Agents writing to this bucket
 }
 
-# EBS Volume for Jenkins Data
+# EFS for Jenkins Data (JENKINS_HOME)
+resource "aws_efs_file_system" "foo" {
+  creation_token = "bfd-${var.env_config.env}-jenkins-data"
+
+  tags = {
+    Name = "bfd-${var.env_config.env}-jenkins-data"
+  }
+}
 
 resource "aws_ebs_volume" "jenkins_data" {
-  availability_zone = data.aws_subnet.selected[0].availability_zone
+  availability_zone = var.env_config.azs
   size              = 1000
   type              = "gp2"
+  encrypted         = true
+  kms_key_id        = data.aws_kms_key.master_key.arn
 
   tags = {
-    Name       = "bfd-mgmt-jenkins-data-master"
+    Name       = "bfd-${var.env_config.env}-jenkins-data"
     cpm_backup = "4HR Daily Weekly Monthly"
   }
 }
 
 # IAM Roles, Profiles and Policies for Packer
```

Comment:
Do you know if any of the packer roles/policies are actually used?  I suspect not unless we are doing something fancy in jenkins to sandbox packer (didn't appear so when I checked)

Reply:
Good question, might be dead / unused. Will verify.

Reply:
I think the removal of this is out of scope. I will take a look at this when testing in mgmt-test and remove if needed.